oak_runtime/src/node/http/user.rs

```diff
@@ -0,0 +1,303 @@
+/// response.
+struct Pipe {
+    request_writer: oak_abi::Handle,
+    request_reader: oak_abi::Handle,
+    response_writer: oak_abi::Handle,
+    response_reader: oak_abi::Handle,
+}
+
+impl Pipe {
+    fn new(
+        runtime: &RuntimeProxy,
+        request_label: &Label,
+        privilege: &NodePrivilege,
+    ) -> Result<Self, ()> {
+        let request_reader_label = Label {
+            confidentiality_tags: request_label.confidentiality_tags.clone(),
+            integrity_tags: privilege
+                .can_endorse_integrity_tags
+                .clone()
+                .into_iter()
+                .collect(),
+        };
+        let response_writer_label = Label {
+            confidentiality_tags: privilege
+                .can_declassify_confidentiality_tags
+                .clone()
+                .into_iter()
+                .collect(),
+            integrity_tags: vec![],
+        };
+
+        // Create a channel for passing HTTP requests to the Oak node. This channel is created with
+        // the label specified by the caller, and the identity of the caller.
+        let (request_writer, request_reader) = runtime
+            .channel_create(&request_reader_label)
+            .map_err(|err| {
+                warn!("could not create HTTP request channel: {:?}", err);
+            })?;
+
+        // Create a channel for receiving HTTP responses from the Oak node. This channel is created
+        // with a label that has the identity of the caller as the confidentiality component.
+        let (response_writer, response_reader) = runtime
+            .channel_create(&response_writer_label)
+            .map_err(|err| {
+                warn!("could not create HTTP response channel: {:?}", err);
+            })?;
+
+        Ok(Pipe {
+            request_writer,
+            request_reader,
+            response_writer,
+            response_reader,
+        })
+    }
+
+    /// Puts the HttpRequest in the `request_sender` channel, ready to be read by the Oak node.
+    fn insert_message(&self, runtime: &RuntimeProxy, request: HttpRequest) -> Result<(), ()> {
+        // Put the HTTP request message inside the per-invocation request channel.
+        let sender = crate::io::Sender::new(WriteHandle {
+            handle: self.request_writer,
+        });
+        sender.send(request, runtime).map_err(|err| {
+            error!(
+                "Couldn't write the request to the HTTP request channel: {:?}",
+                err
+            )
+        })
+    }
+
+    /// Sends an instance of [`HttpInvocation`] to the Oak node. The [`HttpInvocation`] instance
+    /// contains two request-specific channels, one that allows the Oak node to read the request,
+    /// and one on which the Oak node sends the response.
+    fn send_invocation(
+        &self,
+        runtime: &RuntimeProxy,
+        invocation_channel: oak_abi::Handle,
+    ) -> Result<(), ()> {
```

Comment:
Result<(),()> does not seem like a particularly useful type as it does not really convey much information and the results do not seem to be used anywhere.